cpu/cc2538/periph/uart.c

```diff
@@ -308,11 +173,61 @@ void uart_write(uart_t uart, const uint8_t *data, size_t len)
 
 void uart_poweron(uart_t uart)
 {
-    (void) uart;
+    assert(uart < UART_NUMOF);
 
+    /* Turn the clock on first, in case it has been turned off */
+    SYS_CTRL->cc2538_sys_ctrl_unnamed1.RCGCUART |= (1 << uart);
+
+    uart_config[uart].dev->cc2538_uart_ctl.CTLbits.UARTEN = 1;
 }
 
 void uart_poweroff(uart_t uart)
 {
-    (void) uart;
+    assert(uart < UART_NUMOF);
+
+    /* Wait for the TX FIFO to clear */
+    while (uart_config[uart].dev->cc2538_uart_fr.FRbits.TXFF) {}
+
+    uart_config[uart].dev->cc2538_uart_ctl.CTLbits.UARTEN = 0;
+
+    /* Turn the clock off afterwards to save power */
+    SYS_CTRL->cc2538_sys_ctrl_unnamed1.RCGCUART &= ~(1 << uart);
 }
+
+static inline void irq_handler(uart_t uart)
+{
+    assert(uart < UART_NUMOF);
+
+    cc2538_uart_t *u = uart_config[uart].dev;
+
+    uint_fast16_t mis;
```

Comment:
any special reason for using the (in the RIOT context very uncommon) `uint_fast16_t` instead of simply `uint16_t` here?